### PR TITLE
Add service check in UserBasedTimezoneDetector

### PR DESCRIPTION
**Pull request by XWB**

### CODE REVIEW

**Comment by OskarStark** [comment]
looks ok to me

**Comment by rande** [comment]
Can you add unit tests ?

**Comment by XWB** [comment]
Travis fails because SonataUserBundle is not compatible with Symfony 3. Should be fixed once https://github.com/sonata-project/SonataIntlBundle/pull/89 is merged.

**Comment by rande** [comment]
@XWB I am currently updating all bundles to get working with SF3.0.. so just wait a bit ;) 

However we need unit test for the current exception you add.

**Comment by XWB** [comment]
I'll add an unit test.

In the meantime, can you tag a 2.2.4 release? I want to get rid of some deprecation warnings that have been resolved in https://github.com/sonata-project/SonataIntlBundle/commit/8a6977a475ac0664e0438e7e84097aacd30c42bd

**Comment by rande** [comment]
@XWB There is no unit test in this PR. Did you push your commits ?

**Comment by XWB** [comment]
@rande Done.

**Comment by rande** [comment]
Can you squash your commits ? Last step ;)

**Comment by XWB** [comment]
Sure, done :)

**Comment by OskarStark** [comment]
thank you @XWB 

@rande can you please check travis?

**Comment by XWB** [comment]
@OskarStark Travis fails because `sonata-project/user-bundle` cannot be installed on the Symfony 3.0 test suite. Seems unrelated to this PR.

**Comment by OskarStark** [comment]
@sonata-project/contributors RTM i think

File: Timezone/UserBasedTimezoneDetector.php [hunk 1]
```diff
@@ -23,7 +23,7 @@
 class UserBasedTimezoneDetector implements TimezoneDetectorInterface
 {
     /**
-     * @var SecurityContextInterface
+     * @var SecurityContextInterface|TokenStorageInterface
      */
     protected $securityContext;
```
Comment (Soullivaneuh) [inline Timezone/UserBasedTimezoneDetector.php]: This property should be deprecated too with getter/setter to be replaced by $tokenStorage.

Comment (rande) [inline Timezone/UserBasedTimezoneDetector.php]: This can be done in another PR

**Comment by rande** [comment]
@XWB Thanks

**Comment by rande** [comment]
@XWB I have reverted this PR due to https://github.com/sonata-project/SonataIntlBundle/issues/97.

Can you send a new one ? thanks.

**Comment by XWB** [comment]
@rande Done https://github.com/sonata-project/SonataIntlBundle/pull/101

I also improved the unit test.

**Pull request merged**